### PR TITLE
Send to phone for App Lab desktop share

### PR DESCRIPTION
**Pull request by joshlory**
Mockup from PM:

Collapsed:

send-to-phone-1

Expanded:

send-to-phone-2

### CODE REVIEW

File: dashboard/app/assets/javascripts/header.js [hunk 1]
```diff
@@ -172,6 +172,45 @@ dashboard.buildHeader = function (stageData, progressData, currentLevelId, userI
   }
 };
 
+dashboard.initSendToPhone = function(selector) {
```
Comment (joshlory) [inline dashboard/app/assets/javascripts/header.js:175]: I'd rather move this logic to the `dashboard.SendToPhone` component, but the jQuery mask plugin wants to manipulate the DOM.

**Comment by Bjvanminnen** [comment]
My initial reaction was also that it would be nice if we could just do all of this in React. What if we found a React equivalent to jquery mask. Maybe https://github.com/insin/react-maskedinput?

**Comment by joshlory** [comment]
I took a look at that library but I was surprised how large it is (4x the size of `jquery.mask.min.js`), and we'd still have both impls [since `/apps` depends on `jquery.mask.min.js`](https://github.com/code-dot-org/code-dot-org/blob/c7247578d7b876183074d520ba09f959f37c412d/apps/src/feedback.js#L729).

I don't think this change makes things worse, but I can fix things up in this PR if you think the work is worth doing now.

File: dashboard/app/assets/javascripts/header.js (hunk 1 above)
Comment (Bjvanminnen) [inline dashboard/app/assets/javascripts/header.js:175]: Can we pass in the element itself here rather than a selector for the parent? (i think so)

**Comment by Bjvanminnen** [comment]
I wonder if we could do something like Brad did in https://github.com/code-dot-org/code-dot-org/pull/4253 where we have a components that doesn't actually use React but is structured similarly, such that (a) it's more declarative and (b) all of it's logic lives in one place instead of across four files.

The way I think this would work...We create a SendToPhone component. On create we attach our event handlers, some of which might change state. We have a `setState` function that modifies `this.state` (and is the only place we modify `this.state`), and then calls `this.render`.

One challenge to this is that we currently use SendToPhone inside of ShareDialogBody, which will strip all click handling, etc. when we call renderToStaticMarkup. I think the way we would want to do this is to have ShareDialogBody return an empty div for SendToPhone, and then after calling renderToStaticMarkup we would render to that div using our React-like class.

I'm not 100% sure this would all work, and it would probably take some time to attempt, but I think that it should work, and I think it could end up being cleaner. If you're interested in exploring this approach, this is something we could potentially pair on on Monday.

**Comment by joshlory** [comment]
After talking to Brent and Alice the plan is to merge this as-is, and open a Pivotal item to track cleanup of the React component.

**Comment by joshlory** [comment]
Opened Pivotal #108493788 to track.

**Pull request merged**